src/stan_math_backend/Locations.ml

```diff
@@ -31,20 +31,27 @@ let prepare_prog (mir : Program.Typed.t) : Program.Numbered.t * state_t =
          (Hashtbl.to_alist label_to_location) ) in
   (mir, location_list)
 
-let pp_globals ppf location_list =
+let gen_globals location_list =
+  let open Cpp in
   let location_list =
     " (found before start of program)"
     :: ( List.filter ~f:(fun x -> x <> Location_span.empty) location_list
        |> List.map ~f:(fun x -> " (in " ^ Location_span.to_string x ^ ")") )
-  in
+    |> List.map ~f:Exprs.literal_string in
   let location_count = List.length location_list in
-  Fmt.pf ppf
-    "@ stan::math::profile_map profiles__;@ static constexpr std::array<const \
-     char*, @[<hov>%d@]> locations_array__ = @ {@[<hov>%a@]};@ "
-    location_count
-    Fmt.(list ~sep:comma (fmt "%S"))
-    location_list
+  let arr_type = Types.const_char_array location_count in
+  [ GlobalVariableDefn
+      (make_variable_defn ~type_:(Type_literal "stan::math::profile_map")
+         ~name:"profiles__" () )
+  ; GlobalVariableDefn
+      (make_variable_defn ~static:true ~constexpr:true ~type_:arr_type
+         ~name:"locations_array__"
+         ~init:(Assignment (ArrayLiteral location_list)) () ) ]
 
-let pp_smeta ppf location_num =
-  if location_num = no_span_num then ()
-  else Fmt.pf ppf "current_statement__ = %d;@;" location_num
+let create_loc_assignment location_num =
```

Comment:
Wouldn't `gen_loc_assignment` be more in line with the other function names? Although I might prefer `assign_loc` or `set_loc`.